core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -385,9 +390,33 @@ private[spark] class MemoryStore(
     }
   }
 
+  def manualClose[T <: MemoryEntry[_]](entry: T): T = {
+    val entryManualCloseTasks = Future {
+      entry match {
+        case e: DeserializedMemoryEntry[_] => e.value.foreach {
+          case o: AutoCloseable =>
+            try {
+              o.close
+            } catch {
+              case NonFatal(e) =>
+                logWarning(s"Got NonFatal exception during remove")
+            }
+          case _ =>
+        }
+        case _ =>
+      }
+    }
+    entryManualCloseTasks.onComplete {
+      case Success(_) =>
+      case Failure(e) => throw e
```

Comment:
To confirm: this is a callback, so throwing an exception doesn't break the Spark application and just gets logged?

Reply:
From prior discussions, I believe that's right. In which case you could just log it